#!/bin/python
PACKAGE = "cr_dai_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("IrFloodLightBrightness_right",    int_t,    1, "Sets the brightness of the IR Flood Light. Limits: up to 1500mA at 30% duty cycle. The duty cycle is controlled by the left camera STROBE, aligned to start of exposure. If the dot projector is also enabled, its lower duty cycle limits take precedence. The emitter is turned off by default", 0,  0, 1500)
gen.add("IrFloodLightBrightness_left",    int_t,    2, "Sets the brightness of the IR Flood Light. Limits: up to 1500mA at 30% duty cycle. The duty cycle is controlled by the left camera STROBE, aligned to start of exposure. If the dot projector is also enabled, its lower duty cycle limits take precedence. The emitter is turned off by default", 0,  0, 1500)
gen.add("IrLaserDotProjectorBrightness_right", int_t,    4, "Sets the brightness of the IR Flood Light. Limits: up to 1500mA at 30% duty cycle. The duty cycle is controlled by the left camera STROBE, aligned to start of exposure. If the dot projector is also enabled, its lower duty cycle limits take precedence. The emitter is turned off by default", 0,  0, 1200)
gen.add("IrLaserDotProjectorBrightness_left", int_t,    8, "Sets the brightness of the IR Flood Light. Limits: up to 1500mA at 30% duty cycle. The duty cycle is controlled by the left camera STROBE, aligned to start of exposure. If the dot projector is also enabled, its lower duty cycle limits take precedence. The emitter is turned off by default", 0,  0, 1200)
gen.add("LogLevel", int_t, 16, "Sets logging level", 0, 3, 6)
exit(gen.generate(PACKAGE, "cr_dai_ros_msgs", "DeviceControl"))